### ISSUE TITLE
Fix error message when migrating accounts.json to key-pairs.json

### ISSUE THREAD

**Issue by NunoAlexandre** [radicle-dev/radicle-registry#420]
- If migrating the storage in `get_or_create_path` fails because `accounts.json` is not readable then we print “Failed to read .../key-pairs.json”. 
- If creating the directory `~/.local/share/radicle-registry-cli` fails we print:
```
Error: Failed to write to the key-pairs file: "/home/nuno/.local/share/radicle-registry-cli/key-pairs.json"
  Caused by: Permission denied (os error 13)
``` 

This means that there are still cases where the user does not have enough information to fix the issue. (In particular it does not resolve the problem that was initially reported)

The only way I see to address this is to add the path information to the error struct when the error is handled from the `std::fs` functions.

_Originally posted by @geigerzaehler in https://github.com/radicle-dev/radicle-registry/pull/416/files_

**Comment by NunoAlexandre** [comment]
Considering the `get_or_create_path` fn:

``` rust
fn get_or_create_path() -> Result<PathBuf, Error> {
    let path_buf = build_path(FILE).expect("build path failed");

    let old_path = build_path("accounts.json").expect("build old_path failed");
    if old_path.exists() {
        println!("=> Migrating the key-pair storage to the latest version...");
        std::fs::rename(old_path, &path_buf).map_err(WritingError::IO).expect("rename failed");
        println!("✓ Done")
    }

    let path = path_buf.as_path();
    if !path.exists() {
        std::fs::write(path, b"{}").map_err(WritingError::IO).expect("write {} failed");
    }

    Ok(path_buf)
}
```

Two cases I tested, running `key-pair list`:

1. Set no permissions at all on the `~/.local/share/radicle-registry-cli` dir

 `if old_path.exists()` doesn't check so it doesn't even attempt to migrate. It then fails on `std::fs::write`, which is executed because `path` doesn't exist (again, because of permissions).

It seems like error-handling wasn't a thing when these libraries were written :thinking: 

The error message is totally sensible for users, since the command is to `list` (read), and it fails to write (sensible internally):

```
Error: Failed to write the key-pairs file: "/home/nuno/.local/share/radicle-registry-cli/key-pairs.json"
  Caused by: Permission denied (os error 13)
```

2. set no permissions at all on the `accounts.json` file

The migration takes place, resulting in `accounts.json` becoming `key-pairs.json`, despite it having no permissions for anything whatsoever. It later fails when trying to `File::open` in `fn list()`, because it doesn't have permissions to open the file.

The error message is good:

```
Error: Failed to read the key-pairs file: "/home/nuno/.local/share/radicle-registry-cli/key-pairs.json"
  Caused by: Permission denied (os error 13)
```

In conclusion, we don't seem to face the reported issue with the migration, although it would be good to have it covered nonetheless. The biggest issue now is report lacking permissions. Instead of checking if a path exists, we should also check whether the user has permissions to read/write it.
